File: 5 Periodo/Padrões/final-project/Factory Method/factoryMethod-typescript/fabricas/FabricaChevrolet.ts
import { Carro } from '../carros/Carro'
import { Celta } from '../carros/Celta'
import { FabricaDeCarro } from './FabricaDeCarro'

export class FabricaChevrolet extends FabricaDeCarro {

	public criarCarro(): Carro {
		return new Celta();
	}

}

File: 5 Periodo/Padrões/final-project/Factory Method/factoryMethod-java/src/br/edu/ifpb/padroes/diegofrazao/fabricas/FabricaChevrolet.java
package br.edu.ifpb.padroes.diegofrazao.fabricas;

import br.edu.ifpb.padroes.diegofrazao.carros.Carro;
import br.edu.ifpb.padroes.diegofrazao.carros.Celta;

public class FabricaChevrolet implements FabricaDeCarro {
    @Override
    public Carro criarCarro() {
        return new Celta();
    }
}
